internal/awsdata/ec2.go

```diff
@@ -82,61 +83,76 @@ func (d *AWSData) loadEC2Instances(region string) {
 	for _, r := range reservations {
 		for _, i := range r.Instances {
 			d.wg.Add(1)
-			go d.processEC2Instance(log, i, accountID, region, partition)
+			go d.processEC2Instance(log, ec2Svc, i, accountID, region, partition)
 		}
 	}
 
 	log.Info("finished processing data")
 }
 
-func (d *AWSData) processEC2Instance(log *logrus.Entry, i *ec2.Instance, accountID string, region string, partition string) {
+func (d *AWSData) processEC2Instance(log *logrus.Entry, ec2Svc ec2iface.EC2API, instance *ec2.Instance, accountID string, region string, partition string) {
 	defer d.wg.Done()
 
 	var name string
-	for _, tag := range i.Tags {
+	for _, tag := range instance.Tags {
 		if *tag.Key == "Name" {
 			name = aws.StringValue(tag.Value)
 		}
 	}
 
+	var public = false
 	var ips []string
 	var macAddresses []string
+	var dnsNames []string
 
-	if aws.StringValue(i.PublicIpAddress) != "" {
-		ips = append(ips, aws.StringValue(i.PublicIpAddress))
+	if aws.StringValue(instance.PublicIpAddress) != "" {
+		ips = append(ips, aws.StringValue(instance.PublicIpAddress))
+		public = true
 	}
 
-	for _, networkInterface := range i.NetworkInterfaces {
+	for _, networkInterface := range instance.NetworkInterfaces {
 		ips = append(ips, aws.StringValue(networkInterface.PrivateIpAddress))
+		for _, ipSet := range networkInterface.PrivateIpAddresses {
```

Comment:
This part is dumping all container IPs running on the instance into this field. Is that the desired behavior? The node is in an EKS cluster using the AWS VPC CNI

Reply:
Hmm, we don't use EKS, so not seeing this myself in my tests. I'll remove it for now and come back to it if it makes sense to include at some point.

Reply:
I don't see any specific mention of how this works with EKS and such under the EC2 docs, but makes sense that they are overlaying on existing EC2 functionality.

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-instance-addressing.html
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/MultipleIP.html

What I did instead was keep this but only include the primary IP address (unless it's already been captured). Does this address the issue for now?

Perhaps when EKS support gets added to `awsinventory`, then this can be better addressed to ensure those EKS containers are showing the IP addresses being used.

Reply:
Yea that makes sense to me. EKS support will include all those IPs in the inventory like you say